src/functions_linearalgebra.jl

```diff
@@ -82,6 +83,20 @@ function Base.getproperty(fact::NamedFactorization{L, T, <:LQ}, d::Symbol) where
     end
 end
 
+# cholesky
+function Base.getproperty(fact::NamedFactorization{L, T, <:Cholesky}, d::Symbol) where {L, T}
+    inner = getproperty(parent(fact), d)
+    n1, n2 = L
+    @assert n1 == n2 # Must be identical given square pdmat?
+    if d === :L
+        return NamedDimsArray{(n1, n2)}(inner)
```

Comment:
For cholesky, I assume that both the lower triangular and upper triangular being square matricies would correspond to the same axis.

Reply:
Yes, that would be my assumption as well. That being said, there's nothing stopping someone from constructing a factorization type with a square matrix, but having different names. Should probably replace the `@assert` with a `DimensionMismatch` error though.

Reply:
Could probably also rewrite this block as:

```
return d in (:L, :U) ? NamedDimsArray{(n1, n2)}(inner) : inner
```

Reply:
I realised actually we don't want the dimcheck at the point of getproperty, but at the point of constructing the `NamedFactoriztion`. I've added a new constructor for this